Compilation error on master using MacOS

The last PR introduce a compilation error due to namespaces ambiguities:
https://github.com/STORM-IRIT/Radium-Releases/runs/1539840285?check_suite_focus=true#step:7:106

Well, I do not understand the error. 
First, After modifying the file to fix the issue, when I compile on my machine (MacOs 10.14.6, AppleClang 10.0.1.10010046), all works fine.

BTW, the CI still stop with the following error at line 22 of file `..../StackTrace.cpp`. 

```
/Users/runner/work/Radium-Engine/Radium-Engine/src/Radium-Engine/src/Core/Utils/StackTrace.cpp:22:1: error: no type named 'string' in namespace 'Ra::Core::Utils::std'; did you mean '::std::string'?
std::string StackTrace( int skip ) {
^~~~~~~~~~~
::std::string
```
But at this line, there is no more this instruction. The file was updated but the CI still use an old one. What happens ? Why ?

@nmellado : is the source cached for the CI ?

--- a/.github/workflows/pull-request-ci.yml
+++ b/.github/workflows/pull-request-ci.yml
@@ -29,11 +29,16 @@ jobs:
        with:
          cached: false
 
-     - name: Clone Radium
+     - name: Prepare directories
        run: |
          mkdir -p install/
-         mkdir -p src && cd src && git clone https://github.com/STORM-IRIT/Radium-Engine.git --single-branch && cd ..
+         mkdir -p src/Radium-Engine
          mkdir -p build/Radium-Engine
+         
+     - name: Checkout remote head
+       uses: actions/checkout@master
+       with:
+         path: src/Radium-Engine
 
      - name: Configure Radium
        run: |

--- a/src/Core/Asset/FileData.hpp
+++ b/src/Core/Asset/FileData.hpp
@@ -12,7 +12,7 @@ namespace Asset {
 class AnimationData;
 class CameraData;
 class GeometryData;
-class VolumeData;
+struct VolumeData;
 class HandleData;
 class LightData;
 

--- a/src/Core/Asset/LightData.hpp
+++ b/src/Core/Asset/LightData.hpp
@@ -181,28 +181,33 @@ class RA_CORE_API LightData : public AssetData
   public:
     Core::Utils::Color m_color;
 
+    struct DirLight {
+        Eigen::Matrix<Scalar, 3, 1> direction;
+    };
+    struct PointLight {
+        Eigen::Matrix<Scalar, 3, 1> position;
+        LightAttenuation attenuation;
+    };
+    struct SpotLight {
+        Eigen::Matrix<Scalar, 3, 1> position;
+        Eigen::Matrix<Scalar, 3, 1> direction;
+        Scalar innerAngle;
+        Scalar outerAngle;
+        LightAttenuation attenuation;
+    };
+    struct AreaLight {
+        // TODO : this representation is usefull but might be improved
+        Eigen::Matrix<Scalar, 3, 1> position;
+        Eigen::Matrix<Scalar, 3, 3> spatialCovariance;
+        Eigen::Matrix<Scalar, 3, 3> normalCovariance;
+        LightAttenuation attenuation;
+    };
+
     union {
-        struct {
-            Eigen::Matrix<Scalar, 3, 1> direction;
-        } m_dirlight;
-        struct {
-            Eigen::Matrix<Scalar, 3, 1> position;
-            LightAttenuation attenuation;
-        } m_pointlight;
-        struct {
-            Eigen::Matrix<Scalar, 3, 1> position;
-            Eigen::Matrix<Scalar, 3, 1> direction;
-            Scalar innerAngle;
-            Scalar outerAngle;
-            LightAttenuation attenuation;
-        } m_spotlight;
-        struct {
-            // TODO : this representation is usefull but might be improved
-            Eigen::Matrix<Scalar, 3, 1> position;
-            Eigen::Matrix<Scalar, 3, 3> spatialCovariance;
-            Eigen::Matrix<Scalar, 3, 3> normalCovariance;
-            LightAttenuation attenuation;
-        } m_arealight;
+        DirLight m_dirlight;
+        PointLight m_pointlight;
+        SpotLight m_spotlight;
+        AreaLight m_arealight;
     };
 };
 

--- a/src/Core/Geometry/Volume.hpp
+++ b/src/Core/Geometry/Volume.hpp
@@ -226,7 +226,7 @@ class RA_CORE_API VolumeGrid : public AbstractDiscreteVolume
     }
 
   private:
-    const ValueType m_defaultValue;
+    ValueType m_defaultValue;
     Container m_data;
 }; // class VolumeGrid
 

--- a/src/Core/Utils/StackTrace.cpp
+++ b/src/Core/Utils/StackTrace.cpp
@@ -1,7 +1,4 @@
 #include <Core/Utils/StackTrace.hpp>
-namespace Ra {
-namespace Core {
-namespace Utils {
 
 #ifndef OS_WINDOWS
 
@@ -10,7 +7,7 @@ namespace Utils {
 #    include <dlfcn.h>    // for dladdr
 #    include <execinfo.h> // for backtrace
 #    include <sstream>
-#    include <string>
+
 #    if defined( COMPILER_CLANG )
 #        pragma clang diagnostic ignored "-Wformat"
 #        pragma clang diagnostic ignored "-Wformat-extra-args"
@@ -19,7 +16,14 @@ namespace Utils {
 #        pragma GCC diagnostic ignored "-Wformat-extra-args"
 #    endif
 
+#endif
+
+namespace Ra {
+namespace Core {
+namespace Utils {
+
 std::string StackTrace( int skip ) {
+#ifndef OS_WINDOWS
     void* callstack[128];
     const int nMaxFrames = sizeof( callstack ) / sizeof( callstack[0] );
     char buf[1024];
@@ -57,15 +61,10 @@ std::string StackTrace( int skip ) {
     }
     if ( nFrames == nMaxFrames ) trace_buf << "  [truncated]\n";
     return trace_buf.str();
-}
-
 #else
-
-inline std::string Backtrace( int skip = 1 ) {
     return "No execution stack trace on windows.";
-}
-
 #endif
+}
 
 } // namespace Utils
 } // namespace Core

--- a/src/Engine/Component/GeometryComponent.hpp
+++ b/src/Engine/Component/GeometryComponent.hpp
@@ -175,7 +175,7 @@ class RA_ENGINE_API PointCloudComponent : public GeometryComponent
     // directly hold a reference to the displayMesh to simplify accesses in handlers
     std::shared_ptr<PointCloud> m_displayMesh {nullptr};
     // The diameter of the splat when rendered
-    float m_splatSize {0.0025};
+    float m_splatSize {0.0025f};
 };
 
 /*-----------------------------------------------------------------------------------------------*/

--- a/src/Engine/Managers/ComponentMessenger/ComponentMessenger.inl
+++ b/src/Engine/Managers/ComponentMessenger/ComponentMessenger.inl
@@ -144,6 +144,7 @@ ComponentMessenger::registerInput( const Entity* entity,
                                    const std::string& id,
                                    const typename CallbackTypes<ReturnType>::Setter& cb ) {
     CORE_ASSERT( entity && comp->getEntity() == entity, "Component not added to entity" );
+    CORE_UNUSED( comp );
     // Will insert a new entity entry if it doesn't exist.
     CallbackMap& entityList = m_entitySetLists[entity];
 

--- a/src/Engine/Managers/ComponentMessenger/ComponentMessenger.inl
+++ b/src/Engine/Managers/ComponentMessenger/ComponentMessenger.inl
@@ -100,11 +100,10 @@ inline bool ComponentMessenger::canRw( const Entity* entity, const std::string& 
 template <typename ReturnType>
 inline void
 ComponentMessenger::registerOutput( const Entity* entity,
-                                    Component* comp,
+                                    Component*,
                                     const std::string& id,
                                     const typename CallbackTypes<ReturnType>::Getter& cb ) {
     CORE_ASSERT( entity && comp->getEntity() == entity, "Component not added to entity" );
-    CORE_UNUSED( comp );
     // Will insert a new entity entry if it doesn't exist.
     CallbackMap& entityList = m_entityGetLists[entity];
 
@@ -120,11 +119,10 @@ ComponentMessenger::registerOutput( const Entity* entity,
 template <typename ReturnType>
 inline void
 ComponentMessenger::registerReadWrite( const Entity* entity,
-                                       Component* comp,
+                                       Component*,
                                        const std::string& id,
                                        const typename CallbackTypes<ReturnType>::ReadWrite& cb ) {
     CORE_ASSERT( entity && comp->getEntity() == entity, "Component not added to entity" );
-    CORE_UNUSED( comp );
     // Will insert a new entity entry if it doesn't exist.
     CallbackMap& entityList = m_entityRwLists[entity];
 
@@ -140,11 +138,10 @@ ComponentMessenger::registerReadWrite( const Entity* entity,
 template <typename ReturnType>
 inline void
 ComponentMessenger::registerInput( const Entity* entity,
-                                   Component* comp,
+                                   Component*,
                                    const std::string& id,
                                    const typename CallbackTypes<ReturnType>::Setter& cb ) {
     CORE_ASSERT( entity && comp->getEntity() == entity, "Component not added to entity" );
-    CORE_UNUSED( comp );
     // Will insert a new entity entry if it doesn't exist.
     CallbackMap& entityList = m_entitySetLists[entity];
 

--- a/src/Engine/Renderer/RenderTechnique/RenderTechnique.cpp
+++ b/src/Engine/Renderer/RenderTechnique/RenderTechnique.cpp
@@ -17,7 +17,7 @@ std::shared_ptr<Ra::Engine::RenderTechnique> RadiumDefaultRenderTechnique {nullp
 RenderTechnique::RenderTechnique() : m_numActivePass {0} {
     for ( auto p = Index( 0 ); p < s_maxNbPasses; ++p )
     {
-        m_activePasses[p]     = std::move( PassConfiguration( ShaderConfiguration(), nullptr ) );
+        m_activePasses[p]     = PassConfiguration( ShaderConfiguration(), nullptr );
         m_passesParameters[p] = nullptr;
     }
 }
@@ -39,7 +39,7 @@ RenderTechnique::~RenderTechnique() = default;
 void RenderTechnique::setConfiguration( const ShaderConfiguration& newConfig,
                                         Core::Utils::Index pass ) {
     m_numActivePass      = std::max( m_numActivePass, pass + 1 );
-    m_activePasses[pass] = std::move( PassConfiguration( newConfig, nullptr ) );
+    m_activePasses[pass] = PassConfiguration( newConfig, nullptr );
     setDirty( pass );
     setConfiguration( pass );
 }

--- a/src/Engine/Renderer/Renderer.hpp
+++ b/src/Engine/Renderer/Renderer.hpp
@@ -426,11 +426,6 @@ class RA_ENGINE_API Renderer
     std::map<std::string, Texture*> m_secondaryTextures;
 
   private:
-    // Qt has the nice idea to bind an fbo before giving you the opengl context,
-    // this flag is used to save it (and render the final screen on it)
-    //  int m_qtPlz;
-    //  int m_qtViewport[4];
-
     // Renderer timings data
     TimerData m_timerData;
 

--- a/src/Engine/Renderer/Renderer.hpp
+++ b/src/Engine/Renderer/Renderer.hpp
@@ -428,8 +428,8 @@ class RA_ENGINE_API Renderer
   private:
     // Qt has the nice idea to bind an fbo before giving you the opengl context,
     // this flag is used to save it (and render the final screen on it)
-    int m_qtPlz;
-    int m_qtViewport[4];
+    //  int m_qtPlz;
+    //  int m_qtViewport[4];
 
     // Renderer timings data
     TimerData m_timerData;

--- a/src/Engine/Renderer/Renderers/DebugRender.cpp
+++ b/src/Engine/Renderer/Renderers/DebugRender.cpp
@@ -182,6 +182,7 @@ void DebugRender::renderPoints( const Core::Matrix4f& viewMatrix,
     uint size = m_points.size();
     if ( 0 == size ) { return; }
 
+    ///\todo use Engine::Mesh
     GLuint vao, vbo;
     glGenVertexArrays( 1, &vao );
 
@@ -237,10 +238,9 @@ void DebugRender::renderMeshes( const Core::Matrix4f& view, const Core::Matrix4f
 
     for ( uint i = 0; i < idx; ++i )
     {
-        Core::Matrix4f model = m_meshes[i].transform.matrix().cast<float>();
-        glUniformMatrix4fv( m_modelLineLoc, 1, GL_FALSE, model.data() );
+        m_lineProg->setUniform( "model", m_meshes[i].transform.matrix() );
         m_meshes[i].mesh->updateGL();
-        // m_meshes[i].mesh->render();
+        m_meshes[i].mesh->render( m_lineProg );
     }
 
     m_meshProg->bind();
@@ -249,10 +249,9 @@ void DebugRender::renderMeshes( const Core::Matrix4f& view, const Core::Matrix4f
 
     for ( uint i = idx; i < m_meshes.size(); ++i )
     {
-        Core::Matrix4f model = m_meshes[i].transform.matrix().cast<float>();
-        glUniformMatrix4fv( m_modelMeshLoc, 1, GL_FALSE, model.data() );
+        m_meshProg->setUniform( "model", m_meshes[i].transform.matrix() );
         m_meshes[i].mesh->updateGL();
-        // m_meshes[i].mesh->render();
+        m_meshes[i].mesh->render( m_meshProg );
     }
 
     m_meshes.clear();

--- a/src/Engine/Renderer/Renderers/DebugRender.hpp
+++ b/src/Engine/Renderer/Renderers/DebugRender.hpp
@@ -95,17 +95,6 @@ class RA_ENGINE_API DebugRender final
     const ShaderProgram* m_pointProg {nullptr};
     const ShaderProgram* m_meshProg {nullptr};
 
-    uint m_modelLineLoc;
-    uint m_viewLineLoc;
-    uint m_projLineLoc;
-
-    uint m_modelMeshLoc;
-    uint m_viewMeshLoc;
-    uint m_projMeshLoc;
-
-    uint m_viewPointLoc;
-    uint m_projPointLoc;
-
     std::vector<Line> m_lines;
     std::vector<DbgMesh> m_meshes;
 

--- a/src/GuiBase/Timeline/TimelineScrollArea.cpp
+++ b/src/GuiBase/Timeline/TimelineScrollArea.cpp
@@ -154,7 +154,7 @@ void TimelineScrollArea::wheelEvent( QWheelEvent* event ) {
         }
 
         double hScroll = horizontalScrollBar()->value();
-        double x       = event->x();
+        double x       = event->position().x();
         double time    = ( hScroll + x - double( m_zero ) ) / double( m_pixPerSec );
 
         onDrawRuler( newRulerWidth );

--- a/src/GuiBase/Timeline/TimelineTimeScale.cpp
+++ b/src/GuiBase/Timeline/TimelineTimeScale.cpp
@@ -20,7 +20,7 @@ void TimelineTimeScale::paintEvent( QPaintEvent* event ) {
     }
 
     QPainter painter( this );
-    painter.setRenderHint( QPainter::HighQualityAntialiasing );
+    painter.setRenderHint( QPainter::Antialiasing );
 
     painter.drawText( 0, 11, "sec" );
     Scalar pixPerSec = m_scrollArea->getPixPerSec();

--- a/src/GuiBase/TreeModel/TreeModel.cpp
+++ b/src/GuiBase/TreeModel/TreeModel.cpp
@@ -94,7 +94,7 @@ QModelIndex TreeModel::parent( const QModelIndex& child ) const {
 Qt::ItemFlags TreeModel::flags( const QModelIndex& index ) const {
     return index.isValid() && getItem( index )->isValid() && getItem( index )->isSelectable()
                ? QAbstractItemModel::flags( index ) | Qt::ItemIsUserCheckable
-               : Qt::ItemFlags( 0 );
+               : Qt::ItemFlags();
 }
 
 void TreeModel::rebuildModel() {

--- a/src/GuiBase/Viewer/FlightCameraManipulator.hpp
+++ b/src/GuiBase/Viewer/FlightCameraManipulator.hpp
@@ -20,7 +20,7 @@ class RA_GUIBASE_API FlightCameraManipulator : public CameraManipulator,
     explicit FlightCameraManipulator( const CameraManipulator& other );
     virtual ~FlightCameraManipulator();
 
-    KeyMappingManager::Context mappingContext();
+    KeyMappingManager::Context mappingContext() override;
     bool handleMousePressEvent( QMouseEvent* event,
                                 const Qt::MouseButtons& buttons,
                                 const Qt::KeyboardModifiers& modifiers,

--- a/src/GuiBase/Viewer/Gizmo/Gizmo.hpp
+++ b/src/GuiBase/Viewer/Gizmo/Gizmo.hpp
@@ -131,7 +131,7 @@ class Gizmo
         UiSelectionControler( const UiSelectionControler& ) = delete;
         ~UiSelectionControler() override                    = default;
         /// Inherited
-        void updateGL();
+        void updateGL() override;
         /// Swap the state of the controler
         void toggleState();
         /// Set the state of the controler to true

--- a/tests/unittest/Core/observer.cpp
+++ b/tests/unittest/Core/observer.cpp
@@ -31,8 +31,9 @@ TEST_CASE( "Core/Utils/Observable", "[Core][Core/Utils][Observable]" ) {
     int c  = 0;
     A::m_b = 0;
 
-    using Observer  = std::function<void( void )>;
-    using Observer2 = std::function<void( int )>;
+    using Observer = std::function<void( void )>;
+    ///\todo add more tests with
+    // using Observer2 = std::function<void( int )>;
 
     auto bf = std::bind( &A::f, &a );
 

